app_test.go

```diff
@@ -391,134 +391,227 @@ func TestNewApp(t *testing.T) {
+func TestPrivate(t *testing.T) {
 	t.Parallel()
 
-	t.Run("CanUsePrivateAndNonPrivateFromOuterModule", func(t *testing.T) {
-		t.Parallel()
+	testCases := []struct {
+		desc string
 
-		app := fxtest.New(t,
-			Module("SubModule", Invoke(func(a int, b string) {})),
-			Provide(func() int { return 0 }, Private),
-			Provide(func() string { return "" }),
-		)
-		app.RequireStart().RequireStop()
-	})
+		// provide is either a Supply or Provide wrapper around the given int
+		// that allows us to generalize these test cases for both APIs.
+		provide func(int, bool) Option
+	}{
+		{
+			desc: "Provide",
+			provide: func(i int, private bool) Option {
+				opts := []any{func() int { return i }}
+				if private {
+					opts = append(opts, Private)
+				}
+				return Provide(opts...)
+			},
+		},
+		{
+			desc: "Supply",
+			provide: func(i int, private bool) Option {
+				opts := []any{i}
+				if private {
+					opts = append(opts, Private)
+				}
+				return Supply(opts...)
+			},
+		},
+	}
 
-	t.Run("CantUsePrivateFromSubModule", func(t *testing.T) {
-		t.Parallel()
+	for _, tt := range testCases {
+		t.Run(tt.desc, func(t *testing.T) {
+			t.Parallel()
 
-		app := NewForTest(t,
-			Module("SubModule", Provide(func() int { return 0 }, Private)),
-			Invoke(func(a int) {}),
-		)
-		err := app.Err()
-		require.Error(t, err)
-		assert.Contains(t, err.Error(), "missing dependencies for function")
-		assert.Contains(t, err.Error(), "missing type: int")
-	})
+			t.Run("CanUsePrivateFromParentModule", func(t *testing.T) {
+				t.Parallel()
+
+				var invoked bool
+				app := fxtest.New(t,
+					Module("SubModule", Invoke(func(a int, b string) {
+						assert.Equal(t, 0, a)
+						invoked = true
+					})),
+					Provide(func() string { return "" }),
+					tt.provide(0, true),
+				)
+				app.RequireStart().RequireStop()
+				assert.True(t, invoked)
+			})
 
-	t.Run("DifferentModulesCanProvideSamePrivateType", func(t *testing.T) {
-		t.Parallel()
+			t.Run("CannotUsePrivateFromSubModule", func(t *testing.T) {
+				t.Parallel()
 
-		app := fxtest.New(t,
-			Module("SubModuleA",
-				Provide(func() int { return 1 }, Private),
-				Invoke(func(s int) {
-					assert.Equal(t, 1, s)
-				}),
-			),
-			Module("SubModuleB",
-				Provide(func() int { return 2 }, Private),
-				Invoke(func(s int) {
-					assert.Equal(t, 2, s)
-				}),
-			),
-			Provide(func() int { return 3 }),
-			Invoke(func(s int) {
-				assert.Equal(t, 3, s)
-			}),
-		)
-		app.RequireStart().RequireStop()
-	})
+				app := NewForTest(t,
+					Module("SubModule", tt.provide(0, true)),
+					Invoke(func(a int) {}),
+				)
+				err := app.Err()
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), "missing dependencies for function")
+				assert.Contains(t, err.Error(), "missing type: int")
+			})
+
+			t.Run("MultipleModulesSameType", func(t *testing.T) {
+				t.Parallel()
+
+				var invoked int
+				app := fxtest.New(t,
+					Module("SubModuleA",
+						tt.provide(1, true),
```

Comment:
naked parameters are hurting the readability. Let's add some comments. 
https://github.com/uber-go/guide/blob/master/style.md#avoid-naked-parameters

```suggestion
						tt.provide(1, true /* private */),
```